tools/transformer/src/atlasTransformations.js

```diff
@@ -74,39 +72,58 @@ module.exports = function runTransformations(openapi) {
 
 // Temporary transformation until new search version is introduced.
 function searchAPIIssuesTransformation(openapi) {
-  if (openapi.components.schemas.SearchIndexResponse) {
-    const responseParent = openapi.components.schemas.SearchIndexResponse;
-    if (
-      responseParent.properties &&
-      responseParent.properties.latestDefinition
-    ) {
-      if (
-        responseParent.discriminator &&
-        responseParent.discriminator.mapping
-      ) {
-        responseParent.properties.latestDefinition = { oneOf: [] };
-        for (const mappingKey in responseParent.discriminator.mapping) {
-          const ref = responseParent.discriminator.mapping[mappingKey];
-          if (!ref) {
-            continue; // Skip if there's no reference
-          }
-          const reference = resolveOpenAPIReference(openapi, ref);
-          responseParent.properties.latestDefinition.oneOf.push({
-            $ref:
-              reference.allOf &&
-              reference.allOf[1] &&
-              reference.allOf[1].properties &&
-              reference.allOf[1].properties.latestDefinition &&
-              reference.allOf[1].properties.latestDefinition.$ref,
+  const modelsToFix = [
+    {
+      modelObject: openapi.components.schemas.SearchIndexResponse,
+      property: "latestDefinition",
+      // Default class model are not correct
+      newModelName: "BaseSearchIndexResponseLatestDefinition",
+    },
+    {
+      modelObject: openapi.components.schemas.SearchIndexCreateRequest,
+      property: "definition",
+      // Default class model are not correct
+      newModelName: "BaseSearchIndexCreateRequestDefinition",
+    },
+  ];
```

Comment:
This transformation is added now as dedicated search fix. However if we see need we can officially document it and support it going forward. Note that this transformation should be short lived as it only mitigates issue in the current API version.